firestore/src/include/firebase/firestore/settings.h

```diff
@@ -137,13 +139,37 @@ class Settings final {
   void set_ssl_enabled(bool enabled);
 
   /**
+   * Returns a reference to the `LocalCacheSettings` instance
+   * used to configure this SDK.
+   */
+  const LocalCacheSettings& local_cache_settings();
```

Comment:
Consider returning by value (i.e. returning a copy) from `local_cache_settings()` since the lifetime of the returned reference is... complicated. Specifically, the returned reference becomes invalid if there is a subsequent call to `set_local_cache_settings()` or `set_persistence_enabled()` or a few other functions. IMO in this particular instance the clarity and memory safety of returning by value far outweighs any performance benefit of returning by reference.

Reply:
I agree with what you said, but I don't think it's possible to return a copy of `LocalCacheSetting` because it is abstract?

We could consider returning the underlying shared_ptr though..this is not in the approved proposal but seems like a good amendment?

Reply:
Hmm fair enough. Maybe returning a const reference is fine. Consider documenting, though, the situations under which the returned reference becomes invalid.